map/light_framework.hpp

```diff
@@ -0,0 +1,85 @@
+#pragma once
+
+#include "map/user.hpp"
+
+#include "ugc/storage.hpp"
+
+#include "base/assert.hpp"
+
+namespace lightweight
+{
+struct LightFrameworkTest;
+
+enum RequestType
+{
+  Empty = 0u,
+  NumberOfUnsentUGC = 1u << 0,
+  UserAuthStatus = 1u << 1,
+  NumberOfUnsentEdits = 1u << 2,
+  All = NumberOfUnsentUGC | UserAuthStatus | NumberOfUnsentEdits
+};
+
+using RequestTypeMask = unsigned;
+
+class Framework
+{
+public:
+  friend struct LightFrameworkTest;
+
+  Framework(RequestTypeMask request) : m_request(request)
+  {
+    CHECK_NOT_EQUAL(request, Empty, ("Mask is empty"));
+    CHECK_LESS_OR_EQUAL(request, All, ("Incorrect mask type:", request));
```

Comment:
Do we really need the "All" value?
The check in the end will take care of it, and we won't have two separate representations for "All" (one as a RequestType and one as a RequestTypeMask).

Alternatively, see routing/vehicle_mask.hpp where we use ```x``` in the enum and ```1 << x``` in the mask.

Reply:
Ok reasonable